File: Object_oriented_programming/Week2/Point.cpp
#include "Point.hpp"
#include <cmath>

Point::Point(int x, int y)
{
    this->x = x;
    this->y = y;
}

int Point::GetX () const
{
    return x;
}

int Point::GetY () const
{
    return y;
}

double Point::DistanceFrom (const Point& other) const
{
    return std::sqrt (
        (GetX () - other.GetX()) * (GetX () - other.GetX()) + 
        (GetY () - other.GetY()) * (GetY () - other.GetY())
    );
}
std::ostream& operator<< (std::ostream& os, const Point& p)
{
    os << "(" << p.GetX () << ", " << p.GetY () << ")";
    return os;
}

File: Java/Week 2/Point.java
class Point {
	
	double x;
	double y;
	
	public void move (double dx, double dy) {
		this.x += dx;
		this.y += dy;
	}
	
	public void mirror (double cx, double cy) {
		this.x = 2 * cx - this.x;
		this.y = 2 * cy - this.y;
	}
	
	public void mirror (Point other) {
		this.x = 2 * other.x - this.x;
		this.y = 2 * other.y - this.y;
	}
	
	public double distance(Point other) {
		double a = this.x - other.x;
		double b = this.y - other.y;
		
		return Math.sqrt(a*a + b*b);
	}
}
